src/main/java/project_16x16/scene/GameplayScene.java

```diff
@@ -829,6 +674,70 @@ public void loadLevel(String path) { // TODO save camera position/settings.
 			}
 		}
 	}
+	
+	public void displayWorldEdit() {
+		displayGrid();
+		if (applet.mousePressEvent && focusedObject != null) {
+			focusedObject.updateEdit(); // enforce one item selected at once
+		}
+	}
+	
+	public void displayGUISlots() {
+		for (int i = 0; i < 6; i++) {
+			// Display Slot
+			image(slot, 20 * 4 / 2 + 10 + i * (20 * 4 + 10), 20 * 4 / 2 + 10);
+
+			// Display Item
+			PImage img = Tileset.getTile(inventory.get(i));
+			applet.image(img, 20 * 4 / 2 + 10 + i * (20 * 4 + 10), 20 * 4 / 2 + 10, img.width * (float) 0.5,
+					img.height * (float) 0.5);
+
+			// Focus Event
+			if (applet.mousePressEvent) {
+				float x = 20 * 4 / 2 + 10 + i * (20 * 4 + 10);
+				float y = 20 * 4 / 2 + 10;
+				if (applet.getMouseCoordScreen().x > x - (20 * 4) / 2
+						&& applet.getMouseCoordScreen().x < x + (20 * 4) / 2
+						&& applet.getMouseCoordScreen().y > y - (20 * 4) / 2
+						&& applet.getMouseCoordScreen().y < y + (20 * 4) / 2) {
+					editorItem.focus = true;
+					editorItem.setTile(inventory.get(i));
+					editorItem.type = Tileset.getTileType(inventory.get(i));
+				}
+			}
+		}
+	}
+	
+	public void changeMode(String mode) {
+		currentMode = modesMap.get(mode);
+		currentMode.enter();
+	}
```

Comment:
I think this method should take in an enum, rather than a string (to prevent arbitrary modes from being entered). You'd have to change modesMap to be of type `HashMap<?modeEnum?, GameplayMode>` too of course.